src/SimplePie.php

```diff
@@ -1816,13 +1823,20 @@ protected function fetch_data(&$cache)
                 $headers = [
                     'Accept' => 'application/atom+xml, application/rss+xml, application/rdf+xml;q=0.9, application/xml;q=0.8, text/xml;q=0.8, text/html;q=0.7, unknown/unknown;q=0.1, application/unknown;q=0.1, */*;q=0.1',
                 ];
-                $file = $this->registry->create(File::class, [$this->feed_url, $this->timeout, 5, $headers, $this->useragent, $this->force_fsockopen, $this->curl_options]);
+                try {
+                    $file = $this->get_http_client()->request(Client::METHOD_GET, $this->feed_url, $headers);
+                } catch (HttpException $th) {
+                    // If the file connection has an error, set SimplePie::error to that and quit
+                    $this->error = $th->getMessage();
+
+                    return !empty($this->data);
+                }
             }
         }
         $this->status_code = $file->status_code;
 
         // If the file connection has an error, set SimplePie::error to that and quit
-        if (!$file->success && !($file->method & self::FILE_SOURCE_REMOTE === 0 || ($file->status_code === 200 || $file->status_code > 206 && $file->status_code < 300))) {
```

Comment:
Same issue here.

Reply:
I think that everything is right here.
1. If `$file` isn't set, but `$this->file` exists, then `$file =& $this->file;` will be used.
2. If `$file` isn't set and `$this->file` doesn't exist, then `$file` will be set by `$file = $this->get_http_client()->request(...)`.
3. And if `$this->get_http_client()` throws an exception than the method will return `$this->data` and `$file` is not needed at all.